### PR TITLE
`Kokkos::ALL` documentation

### LINKED ISSUE #76: Kokkos::ALL has no documentation
I don't know if it is just too small to get a separate page, but Kokkos::ALL does not have it's own documentation page.

### PR DESCRIPTION
**Pull request by JBludau**
Fixes #76 
I just linked to the docpage if `Kokkos::ALL` was used outside of an example.
`subview` seems to be the only page where this is the case.

### CODE REVIEW

File: docs/source/API/core/utilities/all.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+Defined in header `<Kokkos_Core.hpp>`
+
+```c++
+namespace Kokkos{
+  KokkosInternalType ALL = impl_detail;
+}
+```
+
+Type used as tag to specify the selection of all elements in a dimension.
```
Comment (dalg24) [inline docs/source/API/core/utilities/all.md:12]: `Kokkos::ALL` is a constant of unspecified type that is used to ...

File: docs/source/API/core/utilities/all.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+(KokkosALL)=
+# `Kokkos::ALL`
+
+Defined in header `<Kokkos_Core.hpp>`
+
+```c++
+namespace Kokkos{
+  KokkosInternalType ALL = impl_detail;
```
Comment (dalg24) [inline docs/source/API/core/utilities/all.md:8]:
```suggestion
  constexpr UNSPECIFIED-TYPE ALL = IMPLEMENTATION-DETAIL;
```

File: docs/source/API/core/utilities/all.md [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+## Examples
+
+```c++
+Kokkos::View<double***[5]> a("A",N0,N1,N2);
+
+auto s  = Kokkos::subview(a,
+              std::pair<int,int>(3,15),
+              5,
+              Kokkos::ALL,
+              Kokkos::ALL);
```
Comment (dalg24) [inline docs/source/API/core/utilities/all.md:24]: I think it is unnecessarily complicated I would have gone for a rank-2 view but it is your call

File: docs/source/API/core/utilities/all.md (hunk 1 above)
Comment (JBludau) [reply docs/source/API/core/utilities/all.md]: done in [56bb6e](https://github.com/kokkos/kokkos-core-wiki/pull/132/commits/56bb6edb1d564172c365d27c48249968751039b0)

File: docs/source/API/core/utilities/all.md (hunk 3 above)
Comment (JBludau) [reply docs/source/API/core/utilities/all.md]: I changed it in [bf8683](https://github.com/kokkos/kokkos-core-wiki/pull/132/commits/bf8683e0a822244b42515488c1f4a58afd9736e0) to 3 dims and ditched the `std::pair`. This way we have a `Kokkos::ALL` for one runtime and one compile-time dimension while still giving a reasonable use of `subview`.

**Review by dalg24 — APPROVED** [review]

**Comment by dalg24** [comment]
Actually please add to `source/API/alphabetical.md`

**Pull request merged**